community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveIntCollections.java

```diff
@@ -848,4 +840,33 @@ public static boolean contains( int[] values, int candidate )
             throw new IllegalStateException( "Encountered an already added item:" + item +
                     " when adding items uniquely to a collection:" + collection );
         }
-    }}
+    }
+
+    /**
+     * Deduplicates values in the {@code values} array.
+     *
+     * @param values sorted array of int values.
+     * @return the provided array if no duplicates were found, otherwise a new shorter array w/o duplicates.
```

Comment:
The comment on the old code was lying, indeed it didn't need the array to be sorted and we never sorted that array anywhere. This code is exactly the same code we had before I have deleted it, so I am gonna readd it as it was. The sorting problem can be dealt with later in a separate PR if we think it is important.